saleor/order/calculations.py

```diff
@@ -309,7 +327,7 @@ def _calculate_and_add_tax(
     prices_entered_with_tax: bool,
 ):
     if tax_calculation_strategy == TaxCalculationStrategy.TAX_APP:
-        _recalculate_order_prices(manager, order, lines)
+        _recalculate_order_prices(manager, order, lines, prices_entered_with_tax)
         tax_data = manager.get_taxes_for_order(order)
         _apply_tax_data(order, lines, tax_data)
```

Comment:
What about the undiscounted amounts for TaxApp? I see changes for plugins, but tax app response is handled in `_apply_tax_data`

Reply:
We agreed that it should be handled in separate PR. Task for this is added [#14729](https://github.com/saleor/saleor/issues/14729)